grails-app/services/de/iteratec/osm/system/MissingJobResultCheckService.groovy

```diff
@@ -0,0 +1,103 @@
+package de.iteratec.osm.system
+
+import de.iteratec.osm.measurement.environment.wptserver.JobResultPersisterService
+import de.iteratec.osm.measurement.schedule.Job
+import de.iteratec.osm.result.JobResult
+import de.iteratec.osm.result.JobResultStatus
+import grails.gorm.transactions.Transactional
+import org.joda.time.DateTime
+import org.quartz.CronExpression
+
+@Transactional
+class MissingJobResultCheckService {
+    static final int FIRST_CHECK_DAYS_AGO = 1
+    static final long SCHEDULE_THRESHOLD_MINUTES = 3
+
+    JobResultPersisterService jobResultPersisterService
+
+    def fillMissingJobResults() {
+        Date latestCheckDate = getLatestCheck()?.date
+        int missingResults = 0
+        if(latestCheckDate) {
+            missingResults = fillMissingJobResultsSinceDate( new DateTime(latestCheckDate).minusMinutes(SCHEDULE_THRESHOLD_MINUTES.toInteger()))
+            new MissingJobResultCheck(date: new Date(), missingResults: missingResults).save()
```

Comment:
By creating a new MissingJobResultCheck for each run, you will create 24 records per day. Without an index on the `date` column, accessing this table will get inefficient very quickly. 
As we don't use this data at any other place, you could also just create one record and update it everytime afterwards. This will save us from any performance impact.